src/directives/offense/controllerAttack.ts

```diff
@@ -0,0 +1,81 @@
+import {Directive} from '../Directive';
+import {profile} from '../../profiler/decorator';
+import {ScoutOverlord} from '../../overlords/core/scout';
+import {ControllerAttackerOverlord} from '../../overlords/offense/controllerAttacker';
+import {log} from '../../console/log';
+
+interface DirectiveControllerAttackMemory extends FlagMemory {
+	attackPos: protoPos[];
+	roomName: string;
+	readyAtTick: number;
+}
+
+
+@profile
+export class DirectiveControllerAttack extends Directive {
+
+	static directiveName = 'controllerAttack';
+	static color = COLOR_RED;
+	static secondaryColor = COLOR_YELLOW;
+
+	memory: DirectiveControllerAttackMemory;
+
+	constructor(flag: Flag) {
+		super(flag);
+
+		if (this.room) {
+			if (!this.memory.roomName) {
+				this.memory.roomName = this.room.name;
+			}
+		}
+
+		// TODO: Not have a scout at all times
+		this.overlords.scout = new ScoutOverlord(this);
+		this.overlords.controllerAttack = new ControllerAttackerOverlord(this);
+	}
+
+	get readyAtTick(): number {
+		if (!this.memory.readyAtTick || this.memory.readyAtTick < Game.time) {
+			if (this.room && this.room.controller && this.room.controller.upgradeBlocked) {
+				this.memory.readyAtTick = Game.time + this.room.controller.upgradeBlocked;
+				return this.memory.readyAtTick;
+			} else {
+				if (this.memory.readyAtTick) delete this.memory.readyAtTick;
+				return Game.time;
+			}
+		} else {
+			return this.memory.readyAtTick;
+		}
+	}
+
+	get roomName(): string {
+		return this.memory.roomName;
+	}
+
+	get attackPos(): RoomPosition[] | undefined {
```

Comment:
move to overlord

Reply:
also, change name to `attackPositions` since it's multiple positions, and have it return an empty list rather than undefined if there's no vision